addons/account_payment/models/payment_transaction.py

```diff
@@ -206,8 +206,8 @@ def _create_payment(self, **extra_create_values):
             invoices = self.invoice_ids
```

Comment:
I'd find that more logical to do as follow :thinking: 
```suggestion
            invoices = self.invoice_ids
        invoices = invoices.filtered(lambda inv: inv.state != 'cancel')
```